### PR TITLE
Support empty dataframes, with and without schema info.

### LINKED ISSUE #307: .show on an empty dataframe should return a friendlier output
**Describe the bug**
.show on an empty dataframe should return a friendlier output. It currently errors out inside of the `to_pandas()` function call

### PR DESCRIPTION
**Pull request by xcharleslin**
Closes #307 .

## Before:

_(show(0) on dataframe)_
image

_(show() on empty dataframe with schema)_
image

_(show() on empty dataframe with empty schema)_
image

## After:
image

image

image

### CODE REVIEW

File: daft/runners/partitioning.py [hunk 1]
```diff
@@ -92,7 +92,8 @@ def __post_init__(self) -> None:
                 raise ValueError(f"mismatch of column id: {col_id} vs {tile.column_id}")
 
     def __len__(self) -> int:
-        assert len(self.columns) > 0
```
Comment (xcharleslin) [inline daft/runners/partitioning.py:95]: @samster25 what was this assert meant for? is it okay to return 0 in this case?

Comment (samster25) [reply daft/runners/partitioning.py:95]: Before I was thinking that an vPartition of len 0 would be made up of columns that were len 0, but this makes sense as well. I can think of a potential bug though for compute, what if you are reading from a directory of CSVs and one of the files is header-only. Would this still work?

Comment (xcharleslin) [reply daft/runners/partitioning.py:95]: Not sure I fully understand the potential failure case here :sweat_smile:  if the file is header-only and read into a partition, would it still be correct to return the length as zero?

File: daft/viz/repr.py [hunk 2]
```diff
@@ -97,23 +97,49 @@ def vpartition_repr_html(
...
-    """
+    if len(vpartition) == 0:
+        return f"""
+            <div>
+                {tabulate_html_string}
+                <small>(No rows to show)</small>
+            </div>
+        """
+
+    else:
```
Comment (jaychia) [inline daft/viz/repr.py:136]: nit: no need for an else here

**Review by jaychia — APPROVED** [review]
LGTM! Congrats on your first PR :)

File: daft/runners/partitioning.py (hunk 1 above)
Comment (samster25) [reply daft/runners/partitioning.py:95]: I ended up writing a test for my usecase here: https://github.com/Eventual-Inc/Daft/pull/343
Looks like my edge case doesn't really matter since you can't run compute on those columns since they don't exist.  Everything else looks good! :) Great first PR.

**Review by samster25 — APPROVED** [review]
